src/ProjectAudioManager.cpp

```diff
@@ -1089,10 +1076,18 @@ void ProjectAudioManager::OnCommitRecording()
 
 void ProjectAudioManager::OnSoundActivationThreshold()
 {
-   auto &project = mProject;
    auto gAudioIO = AudioIO::Get();
-   if ( gAudioIO && &project == gAudioIO->GetOwningProject().get() ) {
-      wxTheApp->CallAfter( [this]{ Pause(); } );
+   if (gAudioIO)
+   {
+      bool canStop =  CanStopAudioStream();
+
+      gAudioIO->SetPaused(!gAudioIO->IsPaused());
+
+      if (canStop)
+      {
+         // this is what ::OnPause really boils down to, when we consider that we are not scrubbing
```

Comment:
What "proves" that you aren't scrubbing?  I'll tell you:  This function is only reached if recording is happening, and recording is mutually exclusive with scrubbing.  So comment.

Reply:
I mean, comment that reasoning here.  But also, comment that `OnSoundActivationThreshold` is called only during recording, in AudioIOListener.h.  Make the assumptions you rely on an explicit part of the contract for that interface (even if only informally as comments).

Reply:
I was assuming the same (i.e. that when recording, one can not scrub) but it is good to have a confirmation by you and yes, better write that in a comment.